tests/python/gaia-ui-tests/gaiatest/apps/settings/regions/homescreen_settings.py

```diff
@@ -5,35 +5,65 @@
 from marionette_driver import expected, By, Wait
 
 from gaiatest.apps.base import Base
-
+from gaiatest.apps.system.regions.activities import Activities
 
 class HomescreenSettings(Base):
 
-    _page_locator = (By.ID, 'homescreen')
-    _icon_layout_locator = (By.CSS_SELECTOR, '#homescreen div.icon-dialog')
-    _icon_layout_confirmation_button_locator = (By.CLASS_NAME, "value-option-confirm")
+    _page_locator = (By.ID, 'homescreens')
+    _change_homescreen_page_locator = (By.ID, 'homescreens-list')
+    _change_homescreen_locator = (By.CSS_SELECTOR, '[data-l10n-id="change-home-screen"]')
+    _get_more_homescreen_locator = (By.CSS_SELECTOR, '[data-l10n-id="get-more-home-screens"]')
+
+    _wallpaper_preview_locator = (By.CSS_SELECTOR, '.wallpaper-preview')
+    _wallpaper_pick_locator = (By.CLASS_NAME, 'wallpaper-button')
+    _pick_cancel_button_locator = (By.CSS_SELECTOR,
+                                             '#action-menu-list > button:nth-child(4)')
 
     @property
     def screen_element(self):
         return self.marionette.find_element(*self._page_locator)
 
-    def select_icon_layout(self, value):
+    @property
+    def change_screen_element(self):
+        return self.marionette.find_element(*self._change_homescreen_page_locator)
+
+    def open_change_home_screen(self):
         element = Wait(self.marionette).until(
-            expected.element_present(*self._icon_layout_locator))
+            expected.element_present(*self._change_homescreen_locator))
         Wait(self.marionette).until(expected.element_displayed(element))
         element.tap()
-        self.select(value)
+        Wait(self.marionette).until(expected.element_displayed(*self._change_homescreen_page_locator))
 
-    def open_select_icon_layout(self):
+    def open_get_more_home_screen(self):
         element = Wait(self.marionette).until(
-            expected.element_present(*self._icon_layout_locator))
+            expected.element_present(*self._get_more_homescreen_locator))
         Wait(self.marionette).until(expected.element_displayed(element))
         element.tap()
+        Wait(self.marionette).until(expected.element_displayed(*self._change_homescreen_page_locator))
 
-    def close_select_icon_layout(self):
+    def cancel_get_more_home_screen(self):
         self.marionette.switch_to_frame()
-        self.marionette.find_element(*self._icon_layout_confirmation_button_locator).tap()
+        self.marionette.find_element(*self._pick_cancel_button_locator).tap()
         self.apps.switch_to_displayed_app()
         element = Wait(self.marionette).until(
-            expected.element_present(*self._icon_layout_locator))
+            expected.element_present(*self._change_homescreen_page_locator))
         Wait(self.marionette).until(expected.element_displayed(element))
```

Comment:
This might actually be true even before the _pick_cancel_button_locator element has been tapped.
If that's the case, I guess you might wan to wait until the _pick_cancel_button_locator element is not visible anymore and then switch to the displayed app.